packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -740,11 +789,27 @@ export class GarbageCollector implements IGarbageCollector {
         // Basically, it was written in the current GC version.
         if (result.wasSummaryTracked) {
             this.latestSummaryGCVersion = this.currentGCVersion;
+            this.latestSerializedGCBlobs = this.pendingSerializedGCBlobs;
```

Comment:
The pending state should be reset here as its not relevant anymore and should not be used.